src/appleseed.studio/mainwindow/rendering/renderwidget.cpp

```diff
@@ -422,5 +428,29 @@ void RenderWidget::paintEvent(QPaintEvent* event)
     m_painter.end();
 }
 
+void RenderWidget::dragEnterEvent(QDragEnterEvent* event)
+{
+    if (event->mimeData()->hasFormat("text/plain") && m_project.has_trace_context())
+        event->acceptProposedAction();
+}
+
+void RenderWidget::dragMoveEvent(QDragMoveEvent* event)
+{
+    if (event->pos().x() > pos().x() + width() || event->pos().y() > pos().y() + height())
+        event->ignore();
+    else
+        event->accept();
+}
+
+void RenderWidget::dropEvent(QDropEvent* event)
+{
+    const Vector2d drop_pos = Vector2d(
+        static_cast<double>(event->pos().x()) / width(),
+        static_cast<double>(event->pos().y()) / height());
+    std::string material_name = event->mimeData()->text().toUtf8().constData();
```

Comment:
`event->mimeData()->text().toStdString()` seems a little bit simpler.

Also, make the variable const.